pkg/cmd/workspace/code.go

```diff
@@ -64,8 +67,19 @@ var CodeCmd = &cobra.Command{
 			}
 			workspaceId = workspace.Id
 		} else {
-			workspace, err = apiclient_util.GetWorkspace(args[0])
+			var workspaceName string
+			if _, err := url.ParseRequestURI(args[0]); err == nil {
+				workspaceName = url.PathEscape(args[0])
+			} else {
+				workspaceName = args[0]
+			}
```

Comment:
Everything works now, just one more thing: this check for encoding is unnecessary, you can always encode here. 
Remove code between lines 70-75, and instead of `workspaceName` use `url.PathEscape(args[0])` as a parameter of `GetWorkspace` method.